celery/app/base.py

```diff
@@ -1045,7 +1046,13 @@ def _sig_to_periodic_task_entry(self, schedule, sig,
             'options': dict(sig.options, **opts),
         }
 
-    def _add_periodic_task(self, key, entry):
+    def _add_periodic_task(self, key, entry, name=None):
+        if name is None and key in self._conf.beat_schedule:
+            logger.warning(
+                f"Periodic task key='{key}' shadowed a previous unnamed periodic task."
```

Comment:
is it possible that we could add something like this to docs as well? if thats not needed we can go with this as well

Reply:
Sure! The `add_periodic_task` has no docs, I wrote a first version of it.